src/D2L.Bmx/AwsCredsCreator.cs

```diff
@@ -62,6 +70,32 @@ bool nonInteractive
 			}
 		}
 
-		return await awsClient.GetTokensAsync( samlResponse, selectedRoleData, duration.Value );
+		if( cache &&
+			awsCredentialCache.GetCredentials(
```

Comment:
Moved the AWS credential caching up ``AwsCredsCreator`` as an injectable service. 
1. The caching we are adding here is an implementation detail of BMX, doesn't have anything to do with AWS API access
1. Okta credential caching similarly lives a layer up from Okta API code